### ISSUE TITLE
Compiling Javascript with lineman failed

### ISSUE THREAD

**Issue by GustavoCaso** [exercism/exercism.io#3120]
I'm trying to remove the need for installing `lineman` globally, that way there is no need for running `npm install -g lineman` to been able to use `lineman` inside the `frontend` folder. Since is already declared as a dependency in our `package.json`, we can make use of the scripts section.

``` javascript
{
  "name": "frontend",
  "title": "An HTML/JS/CSS app",
  "version": "0.0.1",
  "private": true,
  "author": {
    "name": "John Doe",
    "company": "ACME, Inc."
  },
  "scripts": {
    "watch": "lineman run",
    "build:production": "lineman build"
  },
  "dependencies": {
    "grunt": "~0.4.1",
    "grunt-ngmin": "~0.0.2",
    "lineman": ">=0.11.1"
  },
  "engines": {
    "node": "~>0.10.5",
    "npm": "~>1.2.18"
  }
}
```

by running `npm run watch` or `npm run build:production` will execute the line specify in the scripts section.

Without the need of installing lineman globally I'm able to run `npm run watch` and lineman will watch for javascripts files, but trying to run `npm run build:production` I get this error: 

``` js
> frontend@0.0.1 build:production /Users/morty/dev/exercism_project/exercism.io/frontend
> lineman build

Running "clean:dist" (clean) task
Cleaning generated...OK

Running "common" task

Running "coffee:compile" (coffee) task
File generated/js/app.coffee.js created.
File generated/js/spec.coffee.js created.
>> Destination (generated/js/spec-helpers.coffee.js) not written because compiled files were empty.

Running "jshint:files" (jshint) task
Warning: Path must be a string. Received null Use --force to continue.

Aborted due to warnings.

npm ERR! Darwin 15.6.0
npm ERR! argv "/Users/morty/.nvm/versions/node/v6.3.1/bin/node" "/Users/morty/.nvm/versions/node/v6.3.1/bin/npm" "run" "build:production"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! frontend@0.0.1 build:production: `lineman build`
npm ERR! Exit status 3
npm ERR!
npm ERR! Failed at the frontend@0.0.1 build:production script 'lineman build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the frontend package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     lineman build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs frontend
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls frontend
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/morty/dev/exercism_project/exercism.io/frontend/npm-debug.log
```

So I decided to installed globally to see if it's failing as well I typed `npm install -g lineman` then went back to the `frontend` folder I try `lineman build` will fail again with:

``` js
Running "clean:dist" (clean) task
Cleaning generated...OK

Running "common" task

Running "coffee:compile" (coffee) task
File generated/js/app.coffee.js created.
File generated/js/spec.coffee.js created.
>> Destination (generated/js/spec-helpers.coffee.js) not written because compiled files were empty.

Running "jshint:files" (jshint) task
Warning: Path must be a string. Received null Use --force to continue.

Aborted due to warnings.
```

I don't know if someone else is experience this kind of errors, but maybe we have revise the building process for javascript assets.

By the way I use `nvm` for managing my nodes versions. Maybe that has something to do.

**Comment by kytrinyx** [comment]
@tejasbubane do you have any ideas here? I know you were messing around with the lineman stuff for a while.

**Comment by tejasbubane** [comment]
@GustavoCaso Not sure what is causing that error, but it is [preferred to install `lineman` globally](https://github.com/linemanjs/lineman/blob/master/package.json#L10).

If we go and `npm install` we get this warning:

```
npm WARN prefer global lineman@0.36.6 should be installed with -g
```

That is why even the [guide](http://linemanjs.com/#getting-started) has a global installation instructions.

**Comment by GustavoCaso** [comment]
@tejasbubane I'm not really an expert on lineman, if the guide say it is better to do it globally, then we should, but when I try to compile I get that error, I will investigate to see if I can fix it.

**Comment by mhelmetag** [comment]
I'm getting hung up at `Running "jshint:files" (jshint) task` because of the same warning as well...

**Comment by tejasbubane** [comment]
```
Running "jshint:files" (jshint) task
Warning: Path must be a string. Received null Use --force to continue.

Aborted due to warnings.
```

@GustavoCaso @mhelmetag Possibly related to https://github.com/jshint/jshint/issues/2922 
I am unable to reproduce the error on my machine. Can you run `lineman build -v` and check if that matches the above issue?

**Comment by mhelmetag** [comment]
@tejasbubane , this is where lineman errors out for me with `lineman build -v`:
```
Running "jshint:files" (jshint) task
Verifying property jshint.files exists in config...OK
Files: app/js/namespace.js, app/js/clipboardcopy.js, app/js/script.js, app/js/scroll.js, app/js/stats.js -> files
Options: force=false, reporterOutput=null, curly, eqeqeq, latedef, newcap, noarg, boss, eqnull, sub, browser
Warning: Path must be a string. Received null Use --force to continue.

Aborted due to warnings.
```

**Comment by tejasbubane** [comment]
@mhelmetag So your error seems to match the one in this [issue](https://github.com/jshint/jshint/issues/2922) and is [fixed in grunt-jshint](https://github.com/gruntjs/grunt-contrib-jshint/commit/a0cd3db0638ad860e562e7e12bfff734aa04ac9f). Upgrading to newer version should fix this. 

`grunt-contrib-jshint` is lineman's dependency. I am not seeing the issue with latest `lineman-0.36.6` which uses [grunt-contrib-jshint version 0.11.0](https://github.com/linemanjs/lineman/blob/0.36.6/package.json#L37). Can you try upgrading?

**Comment by mhelmetag** [comment]
Seems like just installing `grunt-contrib-jshint` fixed this. Looks like it wasn't installed for me. Not sure why but I assume this is the issue...

**Comment by tejasbubane** [comment]
Maybe an older version was installed which got upgraded.

**Comment by GustavoCaso** [comment]
@tejasbubane I confirm the same running `lineman build -v --force` it works. So maybe adding documenting it in the `README`, I'll be happy to do it.

**Comment by tejasbubane** [comment]
`--force` is rather a workaround. If I am not wrong, `lineman` runs `grunt` in force mode which ignores warnings. It is [recommended not to be used in the docs here](http://gruntjs.com/using-the-cli#force-f). So I suggest we should not add this to the docs. If someone comes across the error, its there to see in the error log.

Also, @mhelmetag has confirmed that this is not an issue with the newer version of `grunt-contrib-jshint`.

**Comment by tejasbubane** [comment]
Update: I was able to reproduce the issue on a clean install. Opened an issue with lineman: https://github.com/linemanjs/lineman/issues/391

**Comment by troubalex** [comment]
Travis complained about this as well in #3311. Should we wait until lineman upgrades or adjust the build script in the meantime?

**Comment by tejasbubane** [comment]
@troubalex I just sent a PR to lineman for the issue. Does travis always complain or intermittently? Because I see the build passing.

**Comment by troubalex** [comment]
It passed after I had fixed linter warnings I couldn't see until after I had installed `grunt-contrib-jshint`. Travis only complained about the path if I didn't miss anything.

**Comment by tejasbubane** [comment]
@troubalex My fix has been merged into lineman and new version `0.37.0` has been released. I tried removing `node_modules` entirely and using this version (kinda like fresh installation) - no errors - it works! 😃  
Sent a PR to upgrade lineman in exercism.
